mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -460,31 +477,57 @@ const GraphSidebar = (props: {
                             key={['yAxis'].concat(graphParams.yAxisColumnIDs).join('')}
                             columnIDsMap={props.columnIDsMapArray[selectedSheetIndex]}
                             columnDtypesMap={props.columnDtypesMap}
-                            
+
                             graphType={graphParams.graphType}
                             graphAxis={GraphAxisType.Y_AXIS}
                             selectedColumnIDs={graphParams.yAxisColumnIDs}
                             otherAxisSelectedColumnIDs={graphParams.xAxisColumnIDs}
-                            
+
                             updateAxisData={updateAxisData}
                             mitoAPI={props.mitoAPI}
                         />
+                        {props.sheetDataArray[selectedSheetIndex].numRows > 1000 &&
+                            <Row justify='space-between' align='center'>
+                                <Col>
+                                    <p className='text-header-3' title='Turning on Filter to Safe Size only graphs the first 1000 rows of your dataframe, ensuring that your browser won’t crash. Turning it off, graphs the entire dataframe and might crash your browser.'>
+                                        Filter to safe size
+                                    </p>
+                                </Col>
+                                <Col>
+                                    <Toggle
+                                        value={graphParams.safetyFilter}
+                                        onChange={toggleSafetyFilter}
+                                    />
+                                </Col>
+                            </Row>
+                        }
+                        {props.sheetDataArray[selectedSheetIndex].numRows <= 1000 &&
```

Comment:
Check for undefined here first - I would bet this causes sheet crashes (now or in the future), if we don't do this. 

Making the types better on this array would be sicko mode.